src/header/internals/item.rs

```diff
@@ -82,6 +82,14 @@ impl Item {
         }
         self.typed.get_mut(tid).map(|typed| unsafe { typed.downcast_mut_unchecked() })
     }
+
+    pub fn into_typed<H: Header>(self) -> Option<H> {
```

Comment:
Because a `H` may be unsized, I probably can only return `Option<Box<H>>` here. However, then I'm unsure how I'd be casting the box to a concrete type.